src/app/modules/ix-table2/async-data-provider.ts

```diff
@@ -1,31 +1,20 @@
-import { BehaviorSubject, Observable, Subscription, map } from 'rxjs';
+import { Observable, OperatorFunction } from 'rxjs';
 import { EmptyType } from 'app/enums/empty-type.enum';
-import { ArrayDataProvider } from 'app/modules/ix-table2/array-data-provider';
-
-export class AsyncDataProvider<T> extends ArrayDataProvider<T> {
-  private subscription = new Subscription();
-  readonly emptyType$ = new BehaviorSubject<EmptyType>(EmptyType.Loading);
-
-  get isLoading$(): Observable<boolean> {
-    return this.emptyType$.pipe(map((emptyType) => emptyType === EmptyType.Loading));
-  }
-
-  get isError$(): Observable<boolean> {
-    return this.emptyType$.pipe(map((emptyType) => emptyType === EmptyType.Errors));
-  }
+import { ApiCallDirectory, ApiCallMethod } from 'app/interfaces/api/api-call-directory.interface';
+import { BaseDataProvider } from 'app/modules/ix-table2/base-data-provider';
 
+export class AsyncDataProvider<T> extends BaseDataProvider<T> {
   constructor(
-    private request$: Observable<T[]>,
+    private request$: Observable<ApiCallDirectory[ApiCallMethod]['response']>,
   ) {
     super();
-    this.refresh();
   }
 
-  refresh(): void {
+  load<K>(transformFunction?: () => OperatorFunction<K, T[]>): void {
```

Comment:
Since you can pass an observable in when creating a provider, I don't see the need to `transformFunction`, you can transform the original observable.

Reply:
This was an attempt to reach something we discussed earlier. However, I agree that it's too complex. `transformFunction` is removed now